rosbag2/include/rosbag2/writer.hpp

```diff
@@ -95,13 +95,21 @@ class ROSBAG2_PUBLIC Writer
    */
   virtual void write(std::shared_ptr<SerializedBagMessage> message);
 
+  /**
+   * Check if the current recording bagfile needs to be split and rolled over to new file.
+   *
+   * \return true if the bagfile should be split.
+   */
+  virtual bool should_split_bagfile() const;
```

Comment:
Could it be private?

Reply:
I guess it should be protected if virtual

Reply:
This is now private.